### PR TITLE
layers: Fix vkExportMetalObjects

### LINKED ISSUE #4873: Validation layers null out values returned from vkExportMetalObjectsEXT.
**Describe the Issue**
Not sure if this is the correct place for this bug, but if you enable the validation layers and try to use VK_EXT_metal_objects to get the metal objects, the value for the native metal object is always null. Here is an example segment of code that will reproduce this issue. If I enable validation layers, non of the layers are triggered, but the value in deviceInfo.mtlDevice is always null. If I turn off the validation layers the deviceInfo.mtlDevice value is populated with the correct device.

```c++
if (mVkExportMetalObjectsEXT)
{
    VkExportMetalObjectsInfoEXT objectsInfo{};
    VkExportMetalDeviceInfoEXT deviceInfo{};
    
    deviceInfo.sType = VK_STRUCTURE_TYPE_EXPORT_METAL_DEVICE_INFO_EXT;
    
    objectsInfo.sType = VK_STRUCTURE_TYPE_EXPORT_METAL_OBJECTS_INFO_EXT;
    objectsInfo.pNext = &deviceInfo;
    
    mVkExportMetalObjectsEXT(mDevice, &objectsInfo);
    
    return deviceInfo.mtlDevice;
}
```

**Valid Usage ID**
No validation messages encountered leading up to the issue.

**Environment:**
 - OS: MacOS 13.0.1
 - GPU: Radeon Pro 560X/Intel UHD Graphics 630 1536 MB/
 - SDK or header version if building from repo: 1.3.231.1
 - Options enabled (synchronization, best practices, etc.): VK_LAYER_KHRONOS_validation

**Additional context**
https://github.com/KhronosGroup/MoltenVK/issues/1787

### PR DESCRIPTION
**Pull request by juan-lunarg**
closes #4873

### CODE REVIEW

File: layers/generated/layer_chassis_dispatch.cpp [hunk 1]
```diff
@@ -295,6 +295,72 @@ void WrapPnextChainHandles(ValidationObject *layer_data, const void *pNext) {
 
 #define DISPATCH_MAX_STACK_ALLOCATIONS 32
 
+// The vkExportMetalObjects extension returns data from the driver -- we've created a copy of the pNext chain, so
+// copy the returned data to the caller
+void CopyExportMetalObjects(const void *src_chain, const void *dst_chain) {
```
Comment (juan-lunarg) [inline layers/generated/layer_chassis_dispatch.cpp:300]: I used `CopyCreatePipelineFeedbackData` for precedent of copying back the data to the user and writing some manual code that is appended to generated code.

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -5965,8 +5965,8 @@ bool StatelessValidation::manual_PreCallValidateAllocateMemory(VkDevice device,
         }
 #ifdef VK_USE_PLATFORM_METAL_EXT
         skip |= ExportMetalObjectsPNextUtil(
-            VK_EXPORT_METAL_OBJECT_TYPE_METAL_TEXTURE_BIT_EXT, "VUID-VkMemoryAllocateInfo-pNext-06780",
-            "vkAllocateMemory():", "VK_EXPORT_METAL_OBJECT_TYPE_METAL_TEXTURE_BIT_EXT", pAllocateInfo->pNext);
+            VK_EXPORT_METAL_OBJECT_TYPE_METAL_BUFFER_BIT_EXT, "VUID-VkMemoryAllocateInfo-pNext-06780",
+            "vkAllocateMemory():", "VK_EXPORT_METAL_OBJECT_TYPE_METAL_BUFFER_BIT_EXT", pAllocateInfo->pNext);
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:5969]: Minor error I found when writing the positive tests.

"""
VUID-VkMemoryAllocateInfo-pNext-06780
If the pNext chain includes a [VkExportMetalObjectCreateInfoEXT](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkExportMetalObjectCreateInfoEXT.html) structure, its exportObjectType member must be VK_EXPORT_METAL_OBJECT_TYPE_METAL_BUFFER_BIT_EXT
"""

File: tests/positive/other.cpp [hunk 3]
```diff
@@ -776,3 +776,136 @@ TEST_F(VkPositiveLayerTest, EnumeratePhysicalDeviceGroups) {
     debug_reporter.Destroy(test_instance);
     vk::DestroyInstance(test_instance, nullptr);
 }
+
+#ifdef VK_USE_PLATFORM_METAL_EXT
+TEST_F(VkPositiveLayerTest, ExportMetalObjects) {
```
Comment (juan-lunarg) [inline tests/positive/other.cpp:781]: Code handles all existing cases of MetalObjects:
- device
- queue
- buffer
- texture
- ioReference
- event

And also tests for pNext chaining

**Review by mark-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**